### PR TITLE
feat: cart management

### PR DESCRIPTION
**Pull request by ravikumar0403**
Change log:
Added new features:
- guest login
- Product details page
- Add to cart
- remove from cart
- increase/decrease quantity of products in cart

### CODE REVIEW

File: src/components/CartList.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import { useProducts } from "../context/product-context";
```
Comment (rohanmathur91) [inline src/components/CartList.js:2]: Have an index.js file in the context folder, the way you have done for reducers

File: src/components/CartSummary.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  const calcCartPrice = (cartItems) => {
+    return cartItems.reduce(
+      (price, product) => (price = price + product.price * product.quantity),
+      0
+    );
+  };
+
+  const calcTotalPrice = (cartPrice, discount, deliveryCharges) =>
+    cartPrice - discount + deliveryCharges;
```
Comment (rohanmathur91) [inline src/components/CartSummary.js:15]: You can put this in utils folder

**Review by shilpe26 — APPROVED** [review]
▪️Very neat code and easily readable
▪️Logic building is also good
▪️Good use of useEffect()
👍🏻

File: src/context/product-context.js [hunk 3]
```diff
@@ -1,43 +1,39 @@
...
-        payload: data,
-      });
-    } catch (error) {
-      dispatch({
-        type: "GET_PRODUCTS_FAILURE",
-        payload: error,
-      });
+  useEffect(() => {
+    if (user) {
+      fetchCart(dispatch);
```
Comment (yashpatel1797) [inline src/context/product-context.js:27]: nice way to create a function for API so context looks clean.

File: src/components/CartSummary.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+          <div className="summary-price">
+            <p>Discount</p>
+            <p>₹ 499</p>
+          </div>
+        </li>
+        <li>
+          <div className="summary-price">
+            <p>Delivery Charges</p>
+            <p>₹ 499</p>
+          </div>
```
Comment (yashpatel1797) [inline src/components/CartSummary.js:37]: here you have not changed the code I guess or you have missed it.

**Review by yashpatel1797 — APPROVED** [review]
You have followed all best practices which I am aware of like naming, dived code in modules, components usage and services folder for api

File: src/components/CartSummary.js (hunk 4 above)
Comment (ravikumar0403) [reply src/components/CartSummary.js:37]: I'll add this in upcoming PR's

**Pull request merged**